lib/openam.js

```diff
@@ -196,8 +208,9 @@ OpenAMClient.prototype.validateAccessToken = function (accessToken, realm) {
  */
 OpenAMClient.prototype.getProfile = function (userId, realm, sessionId, cookieName) {
     //noinspection JSValidateTypes
-    return request.get(this.serverUrl + '/json/users/' + userId, {
+    return request.get((this.privateIP ? (this.privateIP) : (this.serverUrl)) + '/json/users/' + userId, {
```

Comment:
since this is repeated many times, it seem like it should be abstracted away so that the duplications can be avoided